godump.go

```diff
@@ -67,35 +72,113 @@ func htmlColorize(code, str string) string {
 	return fmt.Sprintf(`<span style="color:%s">%s</span>`, htmlColorMap[code], str)
 }
 
+// Dumper holds configuration for dumping structured data.
+// It controls depth, item count, and string length limits.
+type Dumper struct {
+	maxDepth     int
+	maxItems     int
+	maxStringLen int
+}
+
+// Option defines a functional option for configuring a Dumper.
+type Option func(*Dumper) *Dumper
+
+// WithMaxDepth allows to control how deep the structure will be dumped.
+// Param n must be 0 or greater or this will be ignored, and default MaxDepth will be 15
+func WithMaxDepth(n int) Option {
+	return func(d *Dumper) *Dumper {
+		if n >= 0 {
+			d.maxDepth = n
+		}
+		return d
+	}
+}
+
+// WithMaxItems allows to control how much items from an array, slice or maps can be printed.
+// Param n must be 0 or greater or this will be ignored, and default MaxItems will be 100
+func WithMaxItems(n int) Option {
+	return func(d *Dumper) *Dumper {
+		if n >= 0 {
+			d.maxItems = n
+		}
```

Comment:
You can now use this

```suggestion
		d.maxItems = max(n, 0)
```

golang.org/x/tools' mordenize would report and fix it.

Reply:
Oh you are right. I was wrong. The value is not 0